caas/kubernetes/provider/k8s.go

```diff
@@ -657,12 +658,23 @@ func (k *kubernetesClient) GetService(appName string, mode caas.DeploymentMode,
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	var result caas.Service
+	var (
+		result caas.Service
+		svc    *core.Service
+	)
 	// We may have the stateful set or deployment but service not done yet.
 	if len(servicesList.Items) > 0 {
-		service := servicesList.Items[0]
-		result.Id = string(service.GetUID())
-		result.Addresses = getSvcAddresses(&service, includeClusterIP)
+		for _, s := range servicesList.Items {
+			// Ignore any headless service for this app.
+			if s.Name == appName || s.Name == appName+"-service" {
```

Comment:
do we have an existing helper function for generating the svc name from appName?

Reply:
There's one for legacy service names (with the juju prefix). There's also a method on the controller stack for the controller name but it's not accessible here. I think the safest option is simply to exclude services that are named "foo-endpoints", which is how the headless services are called. That way we cater for all the other possibilities.